.htaccess

```diff
@@ -1,23 +1,25 @@
 # Original
 # If you modify this file then change the above line to: # Modified
+
 <IfModule mod_rewrite.c>
     RewriteEngine On
+
     # Certain hosts may require the following line.
     # If vanilla is in a subfolder then you need to specify it after the /.
     # (ex. You put Vanilla in /forum so change the next line to: RewriteBase /forum)
     # RewriteBase /
 
-    RewriteCond %{QUERY_STRING} ^p=/?([^&]+)(&(.+))?$
-    RewriteRule ^index\.php index.php/%1?%3 [E=X_REWRITE:1,L]
+    RewriteCond %{QUERY_STRING} ^p=/?([^&]+)(&([^?]+))?$
+    RewriteRule ^index\.php %1?%3 [E=X_REWRITE:1,L]
 
     # The basic rewrite rule.
     RewriteCond %{REQUEST_FILENAME} !-f
-    RewriteCond %{REQUEST_FILENAME} !-d
```

Comment:
`/something/index.php` is not how you'd access that file; you'd type `/something` because webservers should automatically look for the index.php file in the directory when you do this. However, removing this rule blocks that behavior and therefore breaks normal access to the app.

I know for a fact it breaks things because it did so on my webserver the _last_ time that change snuck into master branch.